sympy/polys/numberfields.py

```diff
@@ -837,14 +837,21 @@ def primitive_element(extension, x=None, **args):
         x, cls = sympify(x), Poly
     else:
         x, cls = Dummy('x'), PurePoly
-    if not args.get('ex', False):
-        extension = [ AlgebraicNumber(ext, gen=x) for ext in extension ]
-
-        g, coeffs = extension[0].minpoly.replace(x), [1]
 
+    if not args.get('ex', False):
+        gen, coeffs = extension[0], [1]
+        # XXX when minimal_polynomial is extended to work
+        # with AlgebraicNumbers this test can be removed
+        if isinstance(gen, AlgebraicNumber):
+            g = gen.minpoly.replace(x)
```

Comment:
If you don't supply a 2nd arg for replace, doesn't it just return `gen.minpoly`?

Reply:
Yes, we would expect two arguments:

    g = gen.minpoly.replace(gen.minpoly.gen, x)
 
The single argument version was apparently developed as 'syntactic sugar' to simplify the code:

    def replace(f, x, y=None):

        if y is None:
            if f.is_univariate:
                x, y = f.gen, x

Reply:
Oh, sleight of hand. I read that but missed the fact that x is now f.gen and y is the value passed for x.